src/schedule/ll_schedule.c

```diff
@@ -271,12 +271,10 @@ static void schedule_ll_domain_clear(struct ll_schedule_data *sch,
 		sch->domain->registered[cpu_get_id()] = false;
 
 		/* reschedule if we are the last client */
-		if (atomic_read(&sch->domain->num_clients)) {
-			atomic_sub(&sch->domain->num_clients, 1);
-			if (!atomic_read(&sch->domain->num_clients)) {
-				domain_clear(sch->domain);
-				schedule_ll_clients_reschedule(sch);
-			}
+		if (atomic_read(&sch->domain->num_clients) &&
+		    !atomic_sub(&sch->domain->num_clients, 1)) {
+			domain_clear(sch->domain);
+			schedule_ll_clients_reschedule(sch);
```

Comment:
I feel like this is a bugfix that could be separate.

Reply:
oh, that one is no longer needed after the atomic update.